File: src/main/ceylon/com/diabolicallabs/process/manager/ceylon/kieprocess/service/TaskService.java
package com.diabolicallabs.process.manager.ceylon.kieprocess.service;

import com.redhat.ceylon.compiler.java.metadata.Ceylon;
import com.redhat.ceylon.compiler.java.metadata.TypeInfo;
import com.redhat.ceylon.compiler.java.metadata.TypeParameter;
import com.redhat.ceylon.compiler.java.metadata.TypeParameters;
import com.redhat.ceylon.compiler.java.metadata.Variance;
import com.redhat.ceylon.compiler.java.metadata.Ignore;
import com.redhat.ceylon.compiler.java.metadata.Name;
import com.redhat.ceylon.compiler.java.runtime.model.TypeDescriptor;
import com.redhat.ceylon.compiler.java.runtime.model.ReifiedType;
import ceylon.language.Callable;
import ceylon.language.DocAnnotation$annotation$;
import io.vertx.ceylon.core.Vertx;
import io.vertx.core.json.JsonObject;
import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

@Ceylon(major = 8)
public class TaskService implements ReifiedType {

  @Ignore
  public static final io.vertx.lang.ceylon.ConverterFactory<com.diabolicallabs.process.manager.service.TaskService, TaskService> TO_CEYLON = new io.vertx.lang.ceylon.ConverterFactory<com.diabolicallabs.process.manager.service.TaskService, TaskService>() {
    public io.vertx.lang.ceylon.Converter<com.diabolicallabs.process.manager.service.TaskService, TaskService> converter(final TypeDescriptor... descriptors) {
      return new io.vertx.lang.ceylon.Converter<com.diabolicallabs.process.manager.service.TaskService, TaskService>() {
        public TaskService convert(com.diabolicallabs.process.manager.service.TaskService src) {
          return new TaskService(src);
        }
      };
    }
  };

  @Ignore
  public static final io.vertx.lang.ceylon.Converter<TaskService, com.diabolicallabs.process.manager.service.TaskService> TO_JAVA = new io.vertx.lang.ceylon.Converter<TaskService, com.diabolicallabs.process.manager.service.TaskService>() {
    public com.diabolicallabs.process.manager.service.TaskService convert(TaskService src) {
      return src.delegate;
    }
  };

  @Ignore public static final TypeDescriptor $TypeDescriptor$ = TypeDescriptor.klass(TaskService.class);
  @Ignore private final com.diabolicallabs.process.manager.service.TaskService delegate;

  public TaskService(com.diabolicallabs.process.manager.service.TaskService delegate) {
    this.delegate = delegate;
  }

  @Ignore 
  public TypeDescriptor $getType$() {
    return $TypeDescriptor$;
  }

  @Ignore
  public Object getDelegate() {
    return delegate;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService addComment(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::String") @Name("comment")  ceylon.language.String comment, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    java.lang.String arg_2 = io.vertx.lang.ceylon.ToJava.String.safeConvert(comment);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_3 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.addComment(arg_0, arg_1, arg_2, arg_3));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService claim(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.claim(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService complete(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.json::Object") @Name("data")  ceylon.json.Object data, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.json.JsonObject arg_2 = io.vertx.lang.ceylon.ToJava.JsonObject.safeConvert(data);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_3 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.complete(arg_0, arg_1, arg_2, arg_3));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService delegate(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::String") @Name("newUserId")  ceylon.language.String newUserId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    java.lang.String arg_2 = io.vertx.lang.ceylon.ToJava.String.safeConvert(newUserId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_3 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.delegate(arg_0, arg_1, arg_2, arg_3));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService exit(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.exit(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService fail(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.json::Object") @Name("data")  ceylon.json.Object data, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.json.JsonObject arg_2 = io.vertx.lang.ceylon.ToJava.JsonObject.safeConvert(data);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_3 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.fail(arg_0, arg_1, arg_2, arg_3));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService forward(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::String") @Name("newUserId")  ceylon.language.String newUserId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    java.lang.String arg_2 = io.vertx.lang.ceylon.ToJava.String.safeConvert(newUserId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_3 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.forward(arg_0, arg_1, arg_2, arg_3));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService getContent(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable|ceylon.json::Object)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<io.vertx.core.json.JsonObject>> arg_1 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<io.vertx.core.json.JsonObject>(handler) {
      public Object toCeylon(io.vertx.core.json.JsonObject event) {
        return io.vertx.lang.ceylon.ToCeylon.JsonObject.safeConvert(event);
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.getContent(arg_0, arg_1));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService release(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.release(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService resume(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.resume(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService skip(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.skip(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService start(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.start(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService stop(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.stop(arg_0, arg_1, arg_2));
    return this;
  }

  @TypeInfo("com.diabolicallabs.process.manager.ceylon.kieprocess.service::TaskService")
  public TaskService suspend(
    final @TypeInfo("ceylon.language::Integer") @Name("taskId")  ceylon.language.Integer taskId, 
    final @TypeInfo("ceylon.language::String") @Name("userId")  ceylon.language.String userId, 
    final @TypeInfo("ceylon.language::Anything(ceylon.language::Throwable?)") @Name("handler")  Callable<?> handler) {
    java.lang.Long arg_0 = io.vertx.lang.ceylon.ToJava.Long.safeConvert(taskId);
    java.lang.String arg_1 = io.vertx.lang.ceylon.ToJava.String.safeConvert(userId);
    io.vertx.core.Handler<io.vertx.core.AsyncResult<java.lang.Void>> arg_2 = handler == null ? null : new io.vertx.lang.ceylon.CallableAsyncResultHandler<java.lang.Void>(handler) {
      public Object toCeylon(java.lang.Void event) {
        return null;
      }
    };
    TaskService ret = com.diabolicallabs.process.manager.ceylon.kieprocess.service.TaskService.TO_CEYLON.converter().safeConvert(delegate.suspend(arg_0, arg_1, arg_2));
    return this;
  }

}

File: src/main/resources/kieprocess-js/task_service.js
/*
 * Copyright 2014 Red Hat, Inc.
 *
 * Red Hat licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/** @module kieprocess-js/task_service */
var utils = require('vertx-js/util/utils');
var Vertx = require('vertx-js/vertx');

var io = Packages.io;
var JsonObject = io.vertx.core.json.JsonObject;
var JTaskService = com.diabolicallabs.process.manager.service.TaskService;

/**
 @class
*/
var TaskService = function(j_val) {

  var j_taskService = j_val;
  var that = this;

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param comment {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.addComment = function(taskId, userId, comment, handler) {
    var __args = arguments;
    if (__args.length === 4 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'string' && typeof __args[3] === 'function') {
      j_taskService["addComment(java.lang.Long,java.lang.String,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, comment, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.claim = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["claim(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param data {Object} 
   @param handler {function} 
   @return {TaskService}
   */
  this.complete = function(taskId, userId, data, handler) {
    var __args = arguments;
    if (__args.length === 4 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && (typeof __args[2] === 'object' && __args[2] != null) && typeof __args[3] === 'function') {
      j_taskService["complete(java.lang.Long,java.lang.String,io.vertx.core.json.JsonObject,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, utils.convParamJsonObject(data), function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param newUserId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.delegate = function(taskId, userId, newUserId, handler) {
    var __args = arguments;
    if (__args.length === 4 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'string' && typeof __args[3] === 'function') {
      j_taskService["delegate(java.lang.Long,java.lang.String,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, newUserId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.exit = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["exit(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param data {Object} 
   @param handler {function} 
   @return {TaskService}
   */
  this.fail = function(taskId, userId, data, handler) {
    var __args = arguments;
    if (__args.length === 4 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && (typeof __args[2] === 'object' && __args[2] != null) && typeof __args[3] === 'function') {
      j_taskService["fail(java.lang.Long,java.lang.String,io.vertx.core.json.JsonObject,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, utils.convParamJsonObject(data), function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param newUserId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.forward = function(taskId, userId, newUserId, handler) {
    var __args = arguments;
    if (__args.length === 4 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'string' && typeof __args[3] === 'function') {
      j_taskService["forward(java.lang.Long,java.lang.String,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, newUserId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param handler {function} 
   @return {TaskService}
   */
  this.getContent = function(taskId, handler) {
    var __args = arguments;
    if (__args.length === 2 && typeof __args[0] ==='number' && typeof __args[1] === 'function') {
      j_taskService["getContent(java.lang.Long,io.vertx.core.Handler)"](utils.convParamLong(taskId), function(ar) {
      if (ar.succeeded()) {
        handler(utils.convReturnJson(ar.result()), null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.release = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["release(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.resume = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["resume(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.skip = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["skip(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.start = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["start(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.stop = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["stop(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  /**

   @public
   @param taskId {number} 
   @param userId {string} 
   @param handler {function} 
   @return {TaskService}
   */
  this.suspend = function(taskId, userId, handler) {
    var __args = arguments;
    if (__args.length === 3 && typeof __args[0] ==='number' && typeof __args[1] === 'string' && typeof __args[2] === 'function') {
      j_taskService["suspend(java.lang.Long,java.lang.String,io.vertx.core.Handler)"](utils.convParamLong(taskId), userId, function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
      return that;
    } else throw new TypeError('function invoked with invalid arguments');
  };

  // A reference to the underlying Java delegate
  // NOTE! This is an internal API and must not be used in user code.
  // If you rely on this property your code is likely to break if we change it / remove it without warning.
  this._jdel = j_taskService;
};

/**

 @memberof module:kieprocess-js/task_service
 @param vertx {Vertx} 
 @param address {string} 
 @return {TaskService}
 */
TaskService.createProxy = function(vertx, address) {
  var __args = arguments;
  if (__args.length === 2 && typeof __args[0] === 'object' && __args[0]._jdel && typeof __args[1] === 'string') {
    return utils.convReturnVertxGen(JTaskService["createProxy(io.vertx.core.Vertx,java.lang.String)"](vertx._jdel, address), TaskService);
  } else throw new TypeError('function invoked with invalid arguments');
};

// We export the Constructor function
module.exports = TaskService;